parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -520,6 +523,35 @@ public Optional<LogicalType> visit(LogicalTypeAnnotation.IntervalLogicalTypeAnno
     public Optional<LogicalType> visit(LogicalTypeAnnotation.VariantLogicalTypeAnnotation variantLogicalType) {
       return of(LogicalTypes.VARIANT(variantLogicalType.getSpecVersion()));
     }
+
+    @Override
+    public Optional<LogicalType> visit(LogicalTypeAnnotation.GeometryLogicalTypeAnnotation geometryLogicalType) {
+      GeometryType geometryType = new GeometryType();
+      if (geometryLogicalType.getCrs() != null) {
+        geometryType.setCrs(geometryLogicalType.getCrs());
+      }
+      return of(LogicalType.GEOMETRY(geometryType));
+    }
+
+    @Override
+    public Optional<LogicalType> visit(LogicalTypeAnnotation.GeographyLogicalTypeAnnotation geographyLogicalType) {
+      GeographyType geographyType = new GeographyType();
+      if (geographyLogicalType.getCrs() != null) {
+        geographyType.setCrs(geographyLogicalType.getCrs());
+      }
+      if (geographyLogicalType.getAlgorithm() != null) {
+        try {
+          // Convert from schema.EdgeInterpolationAlgorithm to format.EdgeInterpolationAlgorithm
+          EdgeInterpolationAlgorithm algorithm = EdgeInterpolationAlgorithm.valueOf(
```

Comment:
Can we add a direct conversion method between thrift and parquet EdgeInterpolationAlgorithm in this file? String conversion looks inefficient.